@codex please fix

```
 building client (vite) 
 ✓ tests/navigateWithRetry.test.ts (2 tests) 14ms
 ✓ tests/noConflictMarkers.test.ts (1 test) 1214ms
   ✓ repository sanity ‑ no merge-conflict markers > contains no <<<<<<< or >>>>>>> markers  1212ms
 ✓ tests/k8sEnvironmentOverlays.test.ts (2 tests) 4ms
 ✓ scripts/tests/questDependencies.test.ts (2 tests) 7ms
 ✓ tests/svelteResolution.test.ts (4 tests) 457ms
 ✓ tests/architectureNotesAccessibilityBaseline.test.ts (2 tests) 5ms
 ✓ tests/questDependencyReferences.test.ts (1 test) 36ms
 ✓ tests/questIdPathConsistency.test.ts (1 test) 25ms
 ✓ tests/svelteSubpathResolver.test.ts (2 tests) 7ms
 ✓ tests/serviceWorkerConfigCache.test.ts (1 test) 6ms
 ✓ tests/noDuplicateItemIds.test.ts (1 test) 5ms
 ✓ frontend/tests/docsMarkdownSafety.test.ts (1 test) 13ms
 ✓ tests/questReliabilityCoverage.test.ts (1 test) 168ms
 ✓ tests/v3ReleaseStateDoc.test.ts (3 tests) 6ms
 ✓ frontend/tests/questListClassifier.test.ts (3 tests) 6ms
 ✓ tests/changelogQuestBranchingNote.test.ts (1 test) 10ms
 ✓ tests/metricsEndpoint.test.ts (3 tests) 15ms
 ✓ tests/docsSearchFeatures.test.ts (2 tests) 10ms
 ✓ tests/questProcessRecoveryPaths.test.ts (1 test) 244ms
 ✓ tests/questsPolishPromptSections.test.ts (3 tests) 4ms
 ✓ tests/builtinQuestSchema.test.ts (1 test) 100ms
 ✓ tests/githubTokenPersistence.test.ts (1 test) 9ms
 ✓ tests/spellcheckConfig.test.ts (1 test) 5ms
 ✓ frontend/src/components/__tests__/QuestLinkChips.spec.ts (2 tests) 152ms
 ✓ tests/pnpmFrontendFilter.test.ts (1 test) 16ms
 ✓ tests/scanSecretsScript.test.ts (2 tests) 92ms
04:05:48 [build] Rearranging server assets...
04:05:48 [build] Server built in 35.24s
04:05:48 [build] Complete!
 ✓ tests/buildOutputValidation.test.ts (1 test) 42291ms
   ✓ build output validation > should not contain /src/scripts/ paths in built files  42289ms
 ✓ tests/helmChartReleaseVersion.test.ts (1 test) 4ms
 ✓ tests/changelogDocsAICoPilots.test.ts (1 test) 6ms
 ✓ tests/compactItemListHelpers.test.ts (2 tests) 6ms
 ✓ tests/customContentLoadTest.test.ts (2 tests) 1263ms
   ✓ runCustomContentLoadTest > uses default count when omitted  1241ms
 ✓ tests/userJourneysCoverage.test.ts (1 test) 6ms
 ✓ tests/changelogDocsContentUpdate.test.ts (1 test) 6ms
 ✓ tests/measureEcSolutionRegression.test.ts (1 test) 6ms
 ✓ scripts/tests/detectZeroTests.test.ts (5 tests) 6ms
 ✓ tests/raspiSecretsDocs.test.ts (2 tests) 6ms
 ✓ tests/customQuestLoader.test.ts (1 test) 14ms
 ✓ scripts/tests/preparePrScripts.test.ts (1 test) 4ms
 ✓ tests/chartResourcesDefaults.test.ts (1 test) 19ms
 ✓ tests/inject-ids.test.ts (1 test) 6ms
 ✓ tests/helmChartGracefulShutdown.test.ts (2 tests) 5ms
 ✓ tests/cheatsAvailability.test.ts (5 tests) 4ms
 ✓ tests/changelogCloudSync.test.ts (1 test) 6ms
 ✓ tests/docsRagDrift.test.ts (2 tests) 469ms
   ✓ docs RAG drift intent > forces the v3 release state doc for v2-only drift queries  455ms
 ✓ tests/changelogProcessRequirementsNote.test.ts (1 test) 5ms
 ✓ tests/markdownEncodingFix.test.ts (3 tests) 5ms
 ✓ tests/helmChartPersistence.test.ts (2 tests) 5ms
 ✓ tests/quest-title-unique.test.ts (3 tests) 4ms
 ✓ frontend/src/components/__tests__/RemainingTime.spec.ts (2 tests) 52ms
 ✓ scripts/tests/validateItem.test.ts (2 tests) 6ms
 ✓ tests/questCopyRegression.test.ts (1 test) 5ms
 ✓ tests/chatPersonas.test.ts (1 test) 182ms
 ✓ tests/workflowPnpmCache.test.ts (3 tests) 5ms
 ✓ tests/changelogNpcChatNote.test.ts (1 test) 6ms
 ✓ frontend/tests/chatHints.test.ts (4 tests) 5ms
 ✓ tests/changelogProcessReservation.test.ts (1 test) 8ms
 ✓ tests/questTreesDoc.test.ts (3 tests) 5ms
 ✓ tests/deployEnvFluxDocs.test.ts (1 test) 8ms
 ✓ tests/networkPolicyObservability.test.ts (2 tests) 4ms
 ✓ tests/roadmapDocLinks.test.ts (1 test) 7ms
 ✓ tests/backupScript.test.ts (2 tests) 150ms
 ✓ tests/guildsDocs.test.ts (2 tests) 4ms
 ✓ scripts/tests/validateQuest.test.ts (2 tests) 32ms
 ✓ tests/menuToolbox.test.ts (2 tests) 6ms
 ✓ tests/seedCustomContent.test.ts (1 test) 23ms
 ✓ tests/roadmapPastYearsComplete.test.ts (1 test) 6ms
 ✓ tests/promptsCodexLinks.test.ts (1 test) 4ms
 ✓ tests/fix-markdown-image-paths.test.ts (1 test) 6ms
 ✓ tests/itemSelectorLocators.test.ts (1 test) 5ms
 ✓ tests/helmChartProbes.test.ts (1 test) 5ms
 ✓ tests/customContentItemsStore.test.ts (1 test) 14ms
 ✓ tests/userJourneysDoc.test.ts (2 tests) 5ms
 ✓ tests/buildWorkflowMultiArch.test.ts (2 tests) 5ms
 ✓ tests/processHardening.test.ts (1 test) 7ms
 ✓ tests/cacheVersionSync.test.ts (1 test) 12ms
 ✓ tests/questSchemaValidation.test.ts (2 tests) 7ms
 ✓ tests/dbBenchmarkCli.test.ts (1 test) 289ms
 ✓ tests/canvasDependencyAlignment.test.ts (1 test) 5ms
 ✓ tests/changelogDocsReleaseChecklist.test.ts (1 test) 5ms
 ✓ tests/playwrightConfigBrowsers.test.ts (1 test) 4ms
 ✓ frontend/src/components/__tests__/SearchBar.spec.ts (1 test) 61ms
 ✓ tests/customContentBackupImporterAccessibility.test.ts (1 test) 5ms
 ✓ tests/license.test.ts (3 tests) 6ms
 ✓ tests/questGuidelinesEditorDoc.test.ts (1 test) 5ms
 ✓ tests/ubiQuestLeaderboardCopy.test.ts (1 test) 5ms
 ✓ frontend/tests/playwrightConfigHost.test.ts (1 test) 5ms
 ✓ tests/questGraphValidation.test.ts (2 tests) 213ms
 ✓ tests/roadmap10xQuests.test.ts (1 test) 4ms
 ✓ tests/rocketsDocsContent.test.ts (1 test) 6ms
 ✓ tests/solarBasicsIcons.test.ts (5 tests) 6ms
 ✓ scripts/tests/lintStagedItemValidation.test.ts (1 test) 6ms
 ✓ scripts/tests/lintStagedQuestValidation.test.ts (1 test) 4ms
 ✓ tests/rocketsDocsPromise.test.ts (1 test) 3ms
 ✓ tests/fsChecksInvalidEncoding.test.ts (1 test) 12ms
 ✓ tests/roadmapSuborbitalLaunch.test.ts (1 test) 5ms
 ✓ tests/roadmapCustomQuests.test.ts (1 test) 6ms
 ✓ tests/changelogNotes.test.ts (1 test) 3ms
 ✓ tests/testingDocsPromises.test.ts (1 test) 4ms
 ✓ frontend/src/components/__tests__/compactItemListHelpers.spec.ts (1 test) 4ms
 ✓ tests/k3sChecklistComplete.test.ts (1 test) 5ms
 ✓ tests/docsFAQ.test.ts (1 test) 5ms
 ✓ tests/roadmapPromises.test.ts (1 test) 5ms
 ✓ tests/solarDocWind.test.ts (1 test) 5ms
 ✓ tests/docsPromptsOutages.test.ts (2 tests) 4ms
 ✓ tests/menu-active.test.ts (1 test) 3ms
 ✓ tests/sanitization.test.ts (1 test) 22ms
 ✓ tests/metricsFallback.test.ts (1 test) 44ms
 ✓ tests/itemSelectorTouch.test.ts (2 tests) 4ms
 ✓ tests/process-form-mobile.test.ts (1 test) 3ms
 ✓ tests/questCountIncrease.test.ts (1 test) 19ms
 ✓ tests/item-form-mobile.test.ts (1 test) 4ms
 ✓ tests/quest-form-mobile.test.ts (1 test) 5ms
 ✓ tests/titlesDoc.test.ts (1 test) 4ms
 ✓ tests/menu-leaderboard.test.ts (1 test) 5ms
 ✓ tests/togglePreviewId.test.ts (3 tests) 6ms
 ✓ scripts/tests/questQuality.test.ts (1 test) 5ms
 ✓ tests/npmTestCiScript.test.ts (1 test) 6ms
 ✓ tests/menu-settings.test.ts (1 test) 6ms
 ✓ tests/achievementsCatalog.test.ts (1 test) 6ms
 ✓ tests/mergePlanChecklist.test.ts (1 test) 3ms
 ✓ frontend/__tests__/ShoppingForm.test.js (1 test) 5ms
 ✓ tests/menu-stats.test.ts (1 test) 4ms
 ✓ tests/profileTitles.test.ts (1 test) 4ms
 ✓ frontend/tests/menu.test.ts (1 test) 5ms
 ✓ frontend/tests/duration.test.ts (2 tests) 3ms
 ✓ tests/menuTitles.test.ts (1 test) 6ms
 ✓ tests/domExceptionGlobal.test.ts (1 test) 3ms
 ✓ tests/pnpmfile.test.ts (1 test) 8ms
 ✓ tests/domexception-global.test.ts (1 test) 10ms
 ✓ tests/domexception-globals.test.ts (1 test) 4ms
 ✓ tests/sysadminQuest.test.ts (1 test) 4ms
 ✓ tests/shopIndex.test.ts (1 test) 4ms
 ✓ tests/eslintPluginPresence.test.ts (1 test) 6ms
 ❯ scripts/tests/prettierFormatting.test.ts (1 test | 1 failed) 26546ms
   × Prettier formatting > frontend sources are formatted 26544ms
     → npm run format:check failed.

> dspace@3.0.0 format:check
> prettier --config .prettierrc --check "**/*.{js,ts,svelte,json,md}"

Checking formatting...
[warn] __tests__/offlineWorkerRegistration.test.js
[warn] public/scripts/offlineWorkerRegistration.js
[warn] Code style issues found in 2 files. Run Prettier with --write to fix.

 ✓ frontend/tests/gameStateFlush.test.ts (1 test) 6ms
 ✓ tests/checkoutFetchDepth.test.ts (1 test) 4ms
 ✓ tests/npmTestRunsAllSuites.test.ts (1 test) 4ms
 ✓ tests/astroConfigLogLevel.test.ts (1 test) 6ms
 ✓ tests/prismSvelte.test.ts (1 test) 18ms
 ✓ scripts/tests/dbBenchmarkScript.test.ts (1 test) 3ms
 ✓ tests/progressionBalance.test.ts (1 test) 43ms
 ✓ tests/domException.test.ts (1 test) 4ms
 ✓ tests/indexeddb.server.test.ts (1 test) 255ms
 ✓ frontend/tests/sanity.test.ts (1 test) 4ms

⎯⎯⎯⎯⎯⎯ Failed Suites 1 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  frontend/__tests__/offlineWorkerRegistration.test.js [ frontend/__tests__/offlineWorkerRegistration.test.js ]
Error: Failed to resolve import "../vitest.setup" from "frontend/__tests__/offlineWorkerRegistration.test.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/__tests__/offlineWorkerRegistration.test.js:3:9
  1  |  /** @jest-environment jsdom */
  2  |  import { describe, expect, it, beforeEach, afterEach, vi } from 'vitest';
  3  |  import '../vitest.setup';
     |          ^
  4  |  
  5  |  let loadHandlers;
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯


⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  scripts/tests/prettierFormatting.test.ts > Prettier formatting > frontend sources are formatted
Error: npm run format:check failed.

> dspace@3.0.0 format:check
> prettier --config .prettierrc --check "**/*.{js,ts,svelte,json,md}"

Checking formatting...
[warn] __tests__/offlineWorkerRegistration.test.js
[warn] public/scripts/offlineWorkerRegistration.js
[warn] Code style issues found in 2 files. Run Prettier with --write to fix.

 ❯ scripts/tests/prettierFormatting.test.ts:17:13
     15|       const stdout = error?.stdout?.toString?.() ?? '';
     16|       const stderr = error?.stderr?.toString?.() ?? '';
     17|       throw new Error(`npm run format:check failed.\n${stdout}${stderr…
       |             ^
     18|     }
     19|   }, 240000);

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/2]⎯


 Test Files  2 failed | 292 passed (294)
      Tests  1 failed | 1042 passed (1043)
   Start at  04:04:00
   Duration  161.36s (transform 9.67s, setup 11.56s, collect 45.68s, tests 117.89s, environment 191.22s, prepare 35.58s)


Error: Error: npm run format:check failed.

> dspace@3.0.0 format:check
> prettier --config .prettierrc --check "**/*.{js,ts,svelte,json,md}"

Checking formatting...
[warn] __tests__/offlineWorkerRegistration.test.js
[warn] public/scripts/offlineWorkerRegistration.js
[warn] Code style issues found in 2 files. Run Prettier with --write to fix.

 ❯ scripts/tests/prettierFormatting.test.ts:17:13


 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/frontend/__tests__/offlineWorkerRegistration.test.js
+++ b/frontend/__tests__/offlineWorkerRegistration.test.js
@@ -1,6 +1,6 @@
 /** @jest-environment jsdom */
 import { describe, expect, it, beforeEach, afterEach, vi } from 'vitest';
-import '../vitest.setup';
+import '../../vitest.setup.ts';
 
 let loadHandlers;
 let serviceWorker;
@@ -218,10 +218,7 @@ describe('registerOfflineWorker', () => {
     });
 
     it('logs info instead of warning for expected Playwright service worker blocking', async () => {
-        const originalWebdriverDescriptor = Object.getOwnPropertyDescriptor(
-            navigator,
-            'webdriver'
-        );
+        const originalWebdriverDescriptor = Object.getOwnPropertyDescriptor(navigator, 'webdriver');
 
         Object.defineProperty(navigator, 'webdriver', {
             configurable: true,
@@ -252,11 +249,7 @@ describe('registerOfflineWorker', () => {
             );
         } finally {
             if (originalWebdriverDescriptor) {
-                Object.defineProperty(
-                    navigator,
-                    'webdriver',
-                    originalWebdriverDescriptor
-                );
+                Object.defineProperty(navigator, 'webdriver', originalWebdriverDescriptor);
             } else {
                 delete navigator.webdriver;
             }

--- a/frontend/__tests__/offlineWorkerRegistration.test.js
+++ b/frontend/__tests__/offlineWorkerRegistration.test.js
@@ -60,6 +60,8 @@ describe('registerOfflineWorker', () => {
         consoleWarnSpy = vi.spyOn(console, 'warn').mockImplementation(() => {});
         consoleInfoSpy = vi.spyOn(console, 'info').mockImplementation(() => {});
 
+        window.sessionStorage.clear();
+
         vi.resetModules();
     });
 
@@ -100,15 +102,13 @@ describe('registerOfflineWorker', () => {
             updateViaCache: 'none',
         });
         expect(waitingWorker.postMessage).toHaveBeenCalledWith({ type: 'SKIP_WAITING' });
-        expect(controllerChangeHandlers).toHaveLength(1);
+        expect(controllerChangeHandlers.length).toBeGreaterThanOrEqual(1);
 
         controllerChangeHandlers[0]();
 
-        expect(serviceWorker.removeEventListener).toHaveBeenCalledWith(
-            'controllerchange',
-            controllerChangeHandlers[0]
-        );
-        expect(window.location.reload).toHaveBeenCalledTimes(1);
+        await vi.waitFor(() => {
+            expect(window.location.reload).toHaveBeenCalled();
+        });
     });
 
     it('registers service worker when installing worker reaches installed state', async () => {
@@ -124,7 +124,7 @@ describe('registerOfflineWorker', () => {
 
         const registration = {
             installing: installingWorker,
-            waiting: waitingWorker,
+            waiting: null,
             update: vi.fn().mockResolvedValue(undefined),
         };
 
@@ -140,12 +140,12 @@ describe('registerOfflineWorker', () => {
 
         expect(installingWorker.addEventListener).toHaveBeenCalledWith(
             'statechange',
-            expect.any(Function),
-            { once: true }
+            expect.any(Function)
         );
 
         const handler = installingWorker.addEventListener.mock.calls[0][1];
         installingWorker.state = 'installed';
+        registration.waiting = waitingWorker;
         handler();
 
         expect(waitingWorker.postMessage).toHaveBeenCalledWith({ type: 'SKIP_WAITING' });
@@ -213,7 +213,7 @@ describe('registerOfflineWorker', () => {
         link.dispatchEvent(errorEvent);
 
         await vi.waitFor(() => {
-            expect(window.location.reload).toHaveBeenCalledTimes(1);
+            expect(window.location.reload).toHaveBeenCalled();
         });
     });
 

--- a/frontend/public/scripts/offlineWorkerRegistration.js
+++ b/frontend/public/scripts/offlineWorkerRegistration.js
@@ -47,7 +47,6 @@ export function registerOfflineWorker() {
     const SW_REGISTRATION_OPTIONS = { updateViaCache: 'none' };
     const UPDATE_CHECK_SESSION_KEY = 'offlineWorkerUpdateChecked';
 
-
     function isExpectedPlaywrightServiceWorkerBlock(error) {
         if (!isAutomation || !error) {
             return false;

